test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -48,42 +54,52 @@ public final class PopupMenuLocation {
     public static final String TEXT =
             "Long-long-long-long-long-long-long text in the item-";
     private static volatile boolean action = false;
+    private static Robot robot;
+    private static Frame frame;
+    private static Rectangle currentScreenBounds;
+
 
     public static void main(final String[] args) throws Exception {
+        robot = new Robot();
+        robot.setAutoDelay(200);
         GraphicsEnvironment ge =
                 GraphicsEnvironment.getLocalGraphicsEnvironment();
         GraphicsDevice[] sds = ge.getScreenDevices();
         for (GraphicsDevice sd : sds) {
             GraphicsConfiguration gc = sd.getDefaultConfiguration();
-            Rectangle bounds = gc.getBounds();
-            Point point = new Point(bounds.x, bounds.y);
+            currentScreenBounds = gc.getBounds();
+            Point point = new Point(currentScreenBounds.x, currentScreenBounds.y);
             Insets insets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
-            while (point.y < bounds.y + bounds.height - insets.bottom - SIZE) {
-                while (point.x
-                        < bounds.x + bounds.width - insets.right - SIZE) {
+            while (point.y < currentScreenBounds.y + currentScreenBounds.height - insets.bottom - SIZE) {
+                while (point.x <
+                           currentScreenBounds.x + currentScreenBounds.width - insets.right - SIZE) {
```

Comment:
This might not be the most elegant way to abide by the rule to create a new line for long lines. I believe the line limit is 80 characters, which this still violates. And since it looks like you're trying to fix that here, might as well do it correctly.

Reply:
I agree, the first `while` condition doesn't fit 80-column limit anyway, so the second may violate it too, and I think the readability only improves. Especially if you use `ScreenBounds` instead of `currentScreenBounds`.

Please, correct the positions of the closing braces: all of them are moved four spaces too far from their level, including the one which closes `main` method implementation.